### PR TITLE
vimPlugins.vim-figlet: init at 2022-12-08

### PR DESCRIPTION
**Pull request by cwboden**
## Description of changes
I'm adding the `vim-FIGlet` Vim plugin following [the guide here in the docs](https://ryantm.github.io/nixpkgs/languages-frameworks/vim/).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable _(Not applicable)_:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/vim/plugins/vim-plugin-names [hunk 1]
```diff
@@ -872,6 +872,7 @@ https://github.com/catppuccin/vim/,HEAD,catppuccin-vim
 https://github.com/inkarkat/vim-AdvancedSorters/,,vim-advanced-sorters
 https://github.com/Konfekt/vim-CtrlXA/,,
 https://github.com/konfekt/vim-DetectSpellLang/,,
+https://github.com/fadein/vim-FIGlet/,HEAD,
```
Comment (figsoda) [inline pkgs/applications/editors/vim/plugins/vim-plugin-names:875]:
```suggestion
https://github.com/fadein/vim-figlet/,HEAD,
```

Comment (cwboden) [reply pkgs/applications/editors/vim/plugins/vim-plugin-names:875]: The plugin is named with a capital 'FIG'. Is there a reason we don't want to maintain capitalization like we do for other plugins?

Comment (figsoda) [reply pkgs/applications/editors/vim/plugins/vim-plugin-names:875]: The convention is for attribute names to be in (lower) kebab case, and GitHub URLs are case-insensitive. Some of the other plugins might have been added before the convention was formed.

**Review by figsoda — APPROVED** [review]
Due to #232078, the update script fails to update `generated.nix` to actually add this plugin. I will merge this and open a follow-up PR to add this

**Pull request merged**